src/_common/content/content-editor/controls/mention/autocomplete/controls.ts

```diff
@@ -0,0 +1,223 @@
+					} else {
+						this.left = 'auto';
+					}
+
+					this.query = matches[1];
+					this.updateSuggestions(this.query);
+					this.visible = true;
+
+					return;
+				}
+			}
+		}
+
+		this.visible = false;
+	}
+
+	private async updateSuggestions(query: string) {
+		// Stop any existing queued search timeout.
+		if (this.remoteSuggestionTimeout) {
+			clearTimeout(this.remoteSuggestionTimeout);
+			this.remoteSuggestionTimeout = undefined;
+			this.isLoading = false;
+		}
+
+		this.selectedIndex = 0;
+		// If we already have the search results cached, use those and don't query backend.
+		if (ContentEditorMentionCache.hasResults(query)) {
+			this.users = ContentEditorMentionCache.getResults(query);
+			await this.handleInverted();
+		} else {
+			this.users = [];
+
+			// We set a timeout for 200ms here to not send right away when the user is typing fast.
+			this.remoteSuggestionTimeout = setTimeout(async () => {
+				this.isLoading = true;
+
+				const payload = await Api.sendRequest(
+					'/web/search/mention-suggestions?q=' + encodeURIComponent(query),
+					undefined,
+					{ detach: true }
+				);
+
+				if (payload.users) {
+					const searchUsers = User.populate(payload.users);
+
+					// Add to cache
+					if (!ContentEditorMentionCache.hasResults(query)) {
+						ContentEditorMentionCache.addResults(query, searchUsers);
+					}
+
+					// Only process results if the currently active suggestion is still the one we initiated the search with.
+					if (query === this.query) {
+						this.users = searchUsers;
+					}
+				}
+
+				await this.handleInverted();
+				this.isLoading = false;
+			}, 500);
+		}
+	}
+
+	private async handleInverted() {
+		// If we are inverted, scroll the container down.
+		if (this.isInverted) {
+			await this.$nextTick(); // Need to wait here for the list to get bootstrapped before scrolling it.
+			this.$refs.list.scrollTop = this.$refs.list.scrollHeight;
+		}
+	}
+
+	isSelected(userId: number) {
+		return this.users[this.selectedIndex].id === userId;
+	}
+
+	async onKeyDown(e: KeyboardEvent) {
+		if (this.visible) {
+			if (e.key === 'ArrowDown' && this.selectedIndex < this.users.length - 1) {
+				this.selectedIndex++;
+				e.stopPropagation();
+				e.preventDefault();
+			} else if (e.key === 'ArrowUp' && this.selectedIndex > 0) {
+				this.selectedIndex--;
+				e.stopPropagation();
+				e.preventDefault();
+			} else if (e.key === 'Enter' || e.key === 'Tab') {
+				const userToInsert = this.users[this.selectedIndex];
+				if (userToInsert) {
+					e.stopPropagation();
+					e.preventDefault();
+					this.insertUser(userToInsert);
+				}
+			}
+		}
+	}
+
+	onClickInsert(user: User) {
+		this.insertUser(user);
+	}
+
+	insertUser(user: User) {
+		if (this.visible && this.canShow) {
+			// start - end include the @query text, it gets replaced with the insertText call.
+			const start = this.view.state.selection.from - this.query.length - 1;
+			const end = this.view.state.selection.from;
```

Comment:
Same issue pointed here:
https://github.com/gamejolt/gamejolt/pull/320/files#diff-cfbbb22adc54bb5903ea5273c034a36dR88

This will fail if we intend to fix the edge case for editing a mention word in the middle.